api/tests/mocha/services/users.spec.js

```diff
@@ -674,6 +675,574 @@ describe('Users service', () => {
+        sinon.stub(places, 'getPlace').resolves({
+          _id: 'miami',
+          parent: {
+            _id: 'florida'
+          }
+        });
+
+        const response = await service.createUsers([userData]);
+        chai.expect(response[0].error.message).to.equal('Contact is not within place.');
+        chai.expect(response[0].error.translationKey).to.equal('configuration.user.place.contact');
+      });
+    });
+
+    it('succeeds and returns responses if some users fail to be inserted', async () => {
+      sinon.stub(db.medic, 'get').resolves({});
+      service.__set__('validateNewUsername', sinon.stub().resolves());
+      service.__set__('createPlace', sinon.stub().resolves());
+      service.__set__('createUser', sinon.stub().resolves());
+      service.__set__('createContact', sinon.stub().resolves());
+      service.__set__('storeUpdatedPlace', sinon.stub().resolves());
+      service.__set__('createUserSettings', sinon.stub().resolves());
+      const getPlaceStub = sinon.stub(places, 'getPlace');
+      getPlaceStub.withArgs('user1-contact').resolves({
+        _id: 'user1-place',
+        _rev: 2,
+        name: 'user1-place',
+        parent: 'user1-contact'
+      });
+      getPlaceStub.withArgs('user2-contact').resolves();
+
+      const response = await service.createUsers([
+        {
+          username: 'user1',
+          place: 'user1-place',
+          contact: { parent: 'user1-contact' },
+          type: 'national-manager',
+          password: 'Sup3rSecret!',
+        },
+        {
+          username: 'user2',
+          place: 'user2-place',
+          contact: { parent: 'user2-contact' },
+          type: 'national-manager',
+          password: 'Sup3rSecret!',
+        },
+      ]);
+
+      chai.expect(response[0]).to.deep.equal({});
+      chai.expect(response[1].error.message).to.equal('Contact is not within place.');
+      chai.expect(response[1].error.translationKey).to.equal('configuration.user.place.contact');
+    });
+
+    it('succeeds if contact and place are the same', async () => {
+      const userData = {
+        username: 'x',
+        password: COMPLEX_PASSWORD,
+        place: 'foo',
+        contact: { 'parent': 'x' },
+        type: 'national-manager'
+      };
+      service.__set__('validateNewUsername', sinon.stub().resolves());
+      service.__set__('createPlace', sinon.stub().resolves());
+      service.__set__('createUser', sinon.stub().resolves());
+      service.__set__('createContact', sinon.stub().resolves());
+      service.__set__('storeUpdatedPlace', sinon.stub().resolves());
+      service.__set__('createUserSettings', sinon.stub().resolves());
+      sinon.stub(places, 'getPlace').resolves({ _id: 'foo' });
+      userData.place = 'foo';
+      const response = await service.createUsers([userData]);
+      chai.expect(response).to.deep.equal([{}]);
+    });
+
+    it('succeeds and response contains the user, contact and user settings fields', async () => {
```

Comment:
Please update this test so two users are inserted, so we see how the response looks like when multiple users are successfully created.